src/lib/__test__/combine-variables.test.js

```diff
@@ -0,0 +1,75 @@
+import { combineVariables } from '../combine-variables';
+
+describe('combineVariables', () => {
+  test('works with one object', () => {
+    const variables = { backgroundLow: 'red' };
+    const actual = combineVariables(variables);
+    expect(actual).toEqual(variables);
+  });
+
+  test('works with two objects', () => {
+    const expected = {
+      backgroundLow: 'red',
+      backgroundHigh: 'green'
+    };
+
+    const actual = combineVariables(
+      { backgroundLow: 'red', backgroundHigh: 'pink' },
+      { backgroundHigh: 'green' }
+    );
+
+    expect(actual).toEqual(expected);
+  });
+
+  test('works with nested variables', () => {
+    const expected = {
+      background: {
+        low: 'red',
+        high: 'green'
+      },
+      roads: {
+        major: 'yellow',
+        minor: 'white'
+      }
+    };
+
+    const actual = combineVariables(
+      {
+        background: {
+          low: 'red',
+          high: 'purple'
+        },
+        roads: {
+          major: 'yellow',
+          minor: 'white'
+        }
+      },
+      {
+        background: { high: 'green' } // Just tweak one nested property
+      }
+    );
```

Comment:
Nested variables are supported and you can choose specific nested variables to override, in this case `background.high` is being overridden while the rest of the variables remain the same.